File: App025_AdvancedRecyclerViews/app/src/main/java/com/kes/app025_advancedrecyclerviews/MainActivity.java
package com.kes.app025_advancedrecyclerviews;

import android.content.Intent;
import android.os.Bundle;
import android.view.View;
import android.widget.Button;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

import com.kes.app025_advancedrecyclerviews.activities.CardRecycler.CardRecyclerActivity;
import com.kes.app025_advancedrecyclerviews.activities.MultipleSelectionRecycler.MultipleSelectionActivity;
import com.kes.app025_advancedrecyclerviews.activities.MultipleViewRecycler.MultipleViewTypeActivity;
import com.kes.app025_advancedrecyclerviews.activities.SingleSelectionRecycler.SingleRecyclerActivity;

public class MainActivity extends AppCompatActivity {

    Button btnCardRecycler;
    Button btnSingleSelectRecycler;
    Button btnMultipleSelectionRecycler;
    Button btnMultipleViewRecycler;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.activity_main);
        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main), (v, insets) -> {
            Insets systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars());
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom);
            return insets;
        });

        btnCardRecycler = findViewById(R.id.button_card_recycler);
        btnCardRecycler.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(), CardRecyclerActivity.class);
                startActivity(i);
            }
        });

        btnSingleSelectRecycler = findViewById(R.id.button_single_selection_recycler);
        btnSingleSelectRecycler.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(), SingleRecyclerActivity.class);
                startActivity(i);
            }
        });

        btnMultipleSelectionRecycler = findViewById(R.id.button_multiple_selection_recycler);
        btnMultipleSelectionRecycler.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(), MultipleSelectionActivity.class);
                startActivity(i);
            }
        });

        btnMultipleViewRecycler = findViewById(R.id.button_multiple_view_recycler);
        btnMultipleViewRecycler.setOnClickListener(new View.OnClickListener() {
            @Override
            public void onClick(View v) {
                Intent i = new Intent(getApplicationContext(), MultipleViewTypeActivity.class);
                startActivity(i);
            }
        });



    }
}

File: App034_Kr_DataBinding/app/src/main/java/com/kes/app034_kr_databinding/MainActivity.kt
package com.kes.app034_kr_databinding

import android.os.Bundle
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import androidx.activity.enableEdgeToEdge
import androidx.appcompat.app.AppCompatActivity
import androidx.core.view.ViewCompat
import androidx.core.view.WindowInsetsCompat
import androidx.databinding.DataBindingUtil
import com.kes.app034_kr_databinding.databinding.ActivityMainBinding

class MainActivity : AppCompatActivity() {

    private lateinit var binding: ActivityMainBinding


    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        enableEdgeToEdge()
//        setContentView(R.layout.activity_main)
        binding = DataBindingUtil.setContentView(this, R.layout.activity_main)

        ViewCompat.setOnApplyWindowInsetsListener(findViewById(R.id.main)) { v, insets ->
            val systemBars = insets.getInsets(WindowInsetsCompat.Type.systemBars())
            v.setPadding(systemBars.left, systemBars.top, systemBars.right, systemBars.bottom)
            insets
        }

//        binding.apply {
//            button.setOnClickListener() {
//                val input: String = input.text.toString()
//                output.text = "Hello $input!"
//            }
//        }

        val user = User(1, "qwe", "123")

        binding.user = user







    }
}